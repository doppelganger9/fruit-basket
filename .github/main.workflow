workflow "CI" {
  on = "push"
  resolves = ["run npm start", "run npm test", "run prettier check", "report code coverage to Coveralls"]
}

action "install deps" {
  uses = "actions/npm@master"
  args = "ci"
}

action "run npm test" {
  needs = "install deps"
  uses = "actions/npm@master"
  args = "test"
}

action "run prettier check" {
  needs = "run npm test"
  uses = "actions/npm@master"
  args = "run prettier:check"
}

action "report code coverage to Coveralls" {
  needs = "run npm test"
  uses = "actions/npm@master"
  args = "run coveralls"
}

action "run npm start" {
  needs = "run npm test"
  uses = "actions/npm@master"
  args = "start"
}
