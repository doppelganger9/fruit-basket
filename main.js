#!/usr/bin/env node
const totalPrice = require('./lib/total-price');

console.log(JSON.stringify(totalPrice({
    orderedItemsByType: {
      'Apple': 4,
      'Orange': 3,
      'Watermelon': 5,
    }
}), null, 2));
