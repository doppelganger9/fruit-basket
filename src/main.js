#!/usr/bin/env node
const totalPrice = require('./lib/total-price');

const basket = {
  orderedItemsByType: {
    Apple: 4,
    Orange: 3,
    Watermelon: 5,
  },
};

const text = JSON.stringify(totalPrice(basket), null, 2);

console.log(text);

process.exit(0);