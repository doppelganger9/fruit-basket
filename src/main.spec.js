describe('main', () => {
  it('given nothing, when called, then show result in console.log', () => {
    const spy = jest.spyOn(console, 'log');

    jest.spyOn(process, 'exit').mockImplementationOnce(() => {
      throw new Error('process.exit() was called.')
    });

    expect(() => {
      require('./main.js');
    }).toThrow('process.exit() was called.');
    expect(spy.mock.calls).toMatchSnapshot();
    expect(process.exit).toHaveBeenCalledWith(0);
  });
})