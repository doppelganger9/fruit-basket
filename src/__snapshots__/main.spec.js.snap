// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main given nothing, when called, then show result in console.log 1`] = `
Array [
  Array [
    "{
  \\"orderedItemsByType\\": {
    \\"Apple\\": 4,
    \\"Orange\\": 3,
    \\"Watermelon\\": 5
  },
  \\"freeItems\\": {
    \\"Apple\\": 4
  },
  \\"orderTotal\\": 5.5
}",
  ],
]
`;
